rs/ic_os/config/src/lib.rs

```diff
@@ -31,8 +44,380 @@ fn parse_config_line(line: &str) -> Option<(String, String)> {
 pub fn config_map_from_path(config_file_path: &Path) -> Result<ConfigMap> {
     let file_contents = read_to_string(config_file_path)
         .with_context(|| format!("Error reading file: {}", config_file_path.display()))?;
-    Ok(file_contents
+
+    let normalized_file_contents = file_contents.replace("\r\n", "\n").replace("\r", "\n");
+
+    Ok(normalized_file_contents
         .lines()
         .filter_map(parse_config_line)
         .collect())
 }
+
+// Prefix should have a max length of 19 ("1234:6789:1234:6789")
+// It could have fewer characters though. Parsing as an ip address with trailing '::' should work.
+fn is_valid_ipv6_prefix(ipv6_prefix: &str) -> bool {
+    ipv6_prefix.len() <= 19 && format!("{ipv6_prefix}::").parse::<Ipv6Addr>().is_ok()
+}
+
+pub fn get_config_ini_settings(config_file_path: &Path) -> Result<(NetworkSettings, bool)> {
+    let config_map: ConfigMap = config_map_from_path(config_file_path)?;
+
+    let ipv6_prefix = config_map
+        .get("ipv6_prefix")
+        .map(|prefix| {
+            if !is_valid_ipv6_prefix(prefix) {
+                bail!("Invalid ipv6 prefix: {}", prefix);
+            }
+            Ok(prefix.clone())
+        })
+        .transpose()?;
+
+    // Per PFOPS - ipv6_subnet will never not be 64
+    let ipv6_subnet = 64_u8;
```

Comment:
ipv6_subnet_prefix_length ?

Reply:
There is no ipv6_subnet_prefix_length. I can't give you a good reason why exactly though.

Reply:
I mean, wouldn't that be a more fitting name? I had to reverse engineer from `64` and its usages what the variable meant.

Reply:
Oh I see. I just updated the name to ipv6_prefix_length, but I had to update it in a lot of places. See commit: https://github.com/dfinity/ic/pull/1190/commits/a382c79b12a5c1f6a482265e14fdedcece07fccf

Reply:
Suggestion: just use the types from `ipnet` because it provides everything including serde support for that stuff.  I'm using it for the firewall code.

Reply:
We'll look into this in a follow-up PR. This PR will lay the groundwork to get some formalized config structure and after this we'll do an investigation to refactor and add better typing and hopefully collapse some of these structs.